otherlibs/unix/caml/socketaddr.h

```diff
@@ -71,15 +70,34 @@ extern "C" {
 #define alloc_inet_addr caml_unix_alloc_inet_addr
 #endif /* CAML_BUILDING_UNIX */
 
-extern void caml_unix_get_sockaddr (value mladdr,
-                               union sock_addr_union * addr /*out*/,
-                               socklen_param_type * addr_len /*out*/);
-extern value caml_unix_alloc_sockaddr (union sock_addr_union * addr /*in*/,
-                                  socklen_param_type addr_len,
+extern void caml_unix_get_sockaddr (value vaddr,
+                               struct sockaddr_storage * addr /*out*/,
+                               socklen_t * addr_len /*out*/);
+extern value caml_unix_alloc_sockaddr (struct sockaddr * addr /*in*/,
+                                  socklen_t addr_len,
                                   int close_on_error);
 extern value caml_unix_alloc_inet_addr (struct in_addr * inaddr);
 #define GET_INET_ADDR(v) (*((struct in_addr *) (v)))
 
+/* API compatibility between union sock_addr_union and
+   struct sockaddr_storage. */
+#ifndef __cplusplus
+#define caml_unix_get_sockaddr(vaddr, addr, addr_len)                       \
+  caml_unix_get_sockaddr(                                                   \
+    (vaddr),                                                                \
+    _Generic((addr),                                                        \
+             struct sockaddr_storage *: (addr),                             \
```

Comment:
Could, or should, this be `default: ...` instead? I think the error messages (at least with gcc) will make more sense to the end user who probably doesn’t care that much about the fact that a `_Generic` didn’t find a matching type.
(And similar remark for the other `_Generic`, of course)

Reply:
Good point, thanks. I've applied this.